oak_functions/loader/src/server.rs

```diff
@@ -384,14 +356,38 @@ impl WasmState {
                 OakStatus::ErrInvalidArgs
             });
 
+        // TODO(#2707): Refactor to return OakStatus::ErrInvalidHandle.
+        let extension = self
+            .extensions_indices
+            .iter_mut()
+            .find_map(|(_, extension)| {
+                if extension.get_handle() == handle {
+                    Some(extension)
+                } else {
+                    None
+                }
+            })
+            .expect("Fail to find extension with given handle.");
+
         let result = match request {
-            Ok(request) => extension.invoke(self, args, request)?,
+            Ok(request) => {
+                let response = extension.invoke(request);
+                let response_ptr_ptr: AbiPointer = args.nth_checked(3)?;
+                let response_len_ptr: AbiPointer = args.nth_checked(4)?;
+                match response {
+                    Ok(Some(response)) => {
+                        self.write_extension_result(response, response_ptr_ptr, response_len_ptr)
+                    }
+                    Ok(None) =>
+                    // No response was expected. We write the empty response.
+                    {
+                        self.write_extension_result(vec![], response_ptr_ptr, response_len_ptr)
```

Comment:
This also confirms my point above: why return None if  it's just treated as an empty Vec anyways?

Reply:
Done.